pkg/component/runtime/runtime_comm_test.go

```diff
@@ -0,0 +1,80 @@
+// Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+// or more contributor license agreements. Licensed under the Elastic License;
+// you may not use this file except in compliance with the Elastic License.
+
+package runtime
+
+import (
+	"bytes"
+	"context"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+
+	"github.com/elastic/elastic-agent-client/v7/pkg/client"
+	"github.com/elastic/elastic-agent/internal/pkg/core/authority"
+)
+
+type agentInfoMock struct {
+	agentID  string
+	snapshot bool
+	version  string
+}
+
+func (a agentInfoMock) AgentID() string {
+	return a.agentID
+}
+func (a agentInfoMock) Snapshot() bool {
+	return a.snapshot
+}
+
+func (a agentInfoMock) Version() string {
+	return a.version
+}
+
+func (a agentInfoMock) Headers() map[string]string                          { panic("implement me") }
+func (a agentInfoMock) LogLevel() string                                    { panic("implement me") }
+func (a agentInfoMock) ReloadID(ctx context.Context) error                  { panic("implement me") }
+func (a agentInfoMock) SetLogLevel(ctx context.Context, level string) error { panic("implement me") }
+
+func TestRuntimeComm_WriteStartUpInfo_packageVersion(t *testing.T) {
+	agentInfo := agentInfoMock{
+		agentID:  "NCC-1701",
+		snapshot: true,
+		version:  "8.13.0+build1966-09-6",
+	}
+
+	want := client.AgentInfo{
+		ID:       agentInfo.AgentID(),
+		Version:  agentInfo.Version(),
+		Snapshot: agentInfo.Snapshot(),
+	}
+
+	ca, err := authority.NewCA()
+	require.NoError(t, err, "could not create CA")
+	pair, err := ca.GeneratePair()
+	require.NoError(t, err, "could not create certificate pair from CA")
+
+	c := runtimeComm{
+		listenAddr: "localhost",
+		ca:         ca,
+		name:       "a_name",
+		token:      "a_token",
+		cert:       pair,
+		agentInfo:  agentInfo,
+	}
+
+	buff := bytes.Buffer{}
+	err = c.WriteStartUpInfo(&buff)
+	require.NoError(t, err, "failed to write ConnInfo")
+
+	clientv2, _, err := client.NewV2FromReader(&buff, client.VersionInfo{
+		Name: "TestRuntimeComm_WriteConnInfo",
+		Meta: nil,
+	})
+	require.NoError(t, err, "failed creating V2 client")
+
+	assert.Equal(t, &want, clientv2.AgentInfo(),
+		"package version does not match")
```

Comment:
Nit: We should try to make the text explain the reason for the assertion, something like:
```suggestion
		"agent info returned by client must match what has been written on command input")
```